server/app.py

```diff
@@ -186,6 +206,10 @@ async def leave_channel(channel_id, leaving_user: IUser):
     event = IChannelEvent(type="user_leave", content=user)
     channel.events.append(event)
     channel.users = [u for u in channel.users if u.id != user.id]
+    if len(channel.users) == 0:
+        event = IChannelOperations(type='channel_deleted', content=channel)
+        channel_operation_events.append(event)
+        channel.deleted = True
```

Comment:
Channel stays in list since Python doesn't allow for updating of the state since it was defined outside of the context. Updated channel creation to allow for creation of a channel with the same name as a deleted channel